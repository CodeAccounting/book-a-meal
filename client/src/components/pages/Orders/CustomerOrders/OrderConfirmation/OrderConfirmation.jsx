import React, { Component } from 'react';
import moment from 'moment';
import { Link, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { userPropTypes } from '../../../../../helpers/proptypes';
import getOrderFromLocalStorage from '../../../../../helpers/getOrderFromLocalStorage';
import View from '../../../../shared/View';
import OrderSummary from '../../../../shared/OrderSummary';
import OrderAmount from '../../../../shared/OrderAmount';

/**
 * @exports
 * @class OrderConfirmation
 * @extends Component
 * @classdesc Creates OrderConfirmation Component
 * @returns {JSX} OrderConfirmation Component
 */
class OrderConfirmation extends Component {
  static propTypes = {
    ...userPropTypes,
    logout: PropTypes.func.isRequired,
    addOrder: PropTypes.func.isRequired,
    isFetching: PropTypes.bool.isRequired
  }

  /**
   * @constructor
   * @memberof OrderConfirmation
   * @param {object} props
   * @returns {JSX} OrderConfirmation Component
   */
  constructor(props) {
    super();

    this.state = {
      order: getOrderFromLocalStorage(props.user)
    };
  }

  /**
   * @memberof OrderConfirmation
   * @returns {func} addOrder
  */
  handleSubmit = () => {
    const { order } = this.state;
    const { deliveryAddress, deliveryPhoneNo } = order;

    const newOrder = {
      deliveryAddress,
      deliveryPhoneNo,
      meals: [...order.meals.map(meal => ({ mealId: meal.id, quantity: meal.quantity }))]
    };

    localStorage.removeItem('bookamealorder');

    return order.id ? this.props.editOrder(order.id, newOrder) : this.props.addOrder(newOrder);
  }

  /**
   * @memberof OrderReview
   * @returns {JSX} Order Summary Component
  */
  renderCustomerDetails = () => (
    <div className="customer-details">
      <p>{moment().format('dddd, Do MMMM YYYY  h:mm a')}</p>
      <p>Customer: {this.props.user.firstname}&nbsp;{this.props.user.lastname}</p>
      <p>Phone Number: {this.state.order.deliveryPhoneNo}</p>
      <p>Address: {this.state.order.deliveryAddress}</p>
    </div>
  );

  /**
   * @memberof OrderConfirmation
   * @returns {JSX} OrderConfirmation Component
  */
  render() {
    const { user, logout, isFetching } = this.props;
    const msg = 'Please note that orders CANNOT be modified or canceled after 15 minutes of order placement.';

    if (this.state.order.meals.length === 0) return <Redirect to="/" />;

    return (
      <View user={user} logout={logout} type="orderConfirm" isFetching={isFetching}>
        <div className="order-confirmation">
          <div>
            <p className="text-center" style={{ marginTop: '30px', marginBottom: '0' }}>{msg}</p>
            <br />
            <h2>Please Confirm Your Order Details</h2>
            <hr />
            <OrderSummary meals={this.state.order.meals} />
            <OrderAmount meals={this.state.order.meals} />
            {this.renderCustomerDetails()}
            <br />
            <Link to="/" href="/"><button className="btn btn-sec btn-block">Edit Order</button></Link>
            <br /><br />
            <button className="btn btn-pri btn-block" id="checkout" onClick={this.handleSubmit}>Confirm Order</button>
          </div>
        </div>
      </View>
    );
  }
}

export default OrderConfirmation;
