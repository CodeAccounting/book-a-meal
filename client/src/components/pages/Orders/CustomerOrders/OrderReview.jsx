import React, { Component, Fragment } from 'react';
import { Link, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { userPropTypes } from '../../../../helpers/proptypes';
import getOrderFromLocalStorage from '../../../../helpers/getOrderFromLocalStorage';
import { syncValidate, validateRequiredFields } from '../../../../helpers/validations';
import formErrorCounter from '../../../../helpers/formErrorCount';
import updateLocalStorageOrder from '../../../../helpers/updateLocalStorageOrder';
import View from '../../../shared/View';
import RenderInput from '../../../shared/FormComponents/RenderInput';

/**
 * @exports
 * @class OrderReview
 * @extends Component
 * @classdesc Creates OrderReview Component
 * @returns {JSX} OrderReview Component
 */
class OrderReview extends Component {
  static propTypes = {
    ...userPropTypes,
    logout: PropTypes.func.isRequired,
    isFetching: PropTypes.bool.isRequired
  }

  /**
   * @constructor
   * @memberof OrderReview
   * @param {object} props
   * @returns {JSX} OrderReview Component
   */
  constructor(props) {
    super();

    const storedOrder = getOrderFromLocalStorage(props.user);

    this.state = {
      showOrderSummary: false,
      order: storedOrder,
      errors: {
        deliveryPhoneNo: null,
        deliveryAddress: null
      },
      values: {
        deliveryPhoneNo: props.user.phoneNo,
        deliveryAddress: props.user.address,
      }
    };
  }

  /**
   * @memberof OrderReview
   * @param {object} event
   * @returns {func} this.updateStateFormValues
  */
  handleInputChange = event => this.updateStateFormValues(event)

  /**
   * @memberof OrderReview
   * @param {object} event
   *  @returns {func} this.clearFormError
  */
  handleFocus = (event) => {
    const { name } = event.target;
    this.clearFormError(name);
  }

  /**
   * @memberof OrderReview
   * @param {string} event
   * @returns {func} this.validateField
  */
  handleBlur = event => this.validateField(event.target.name)

  /**
   * @memberof OrderReview
   * @returns {func} this.setState
  */
  toggleOrderSummary = () =>
    this.setState(prevState => ({ showOrderSummary: !prevState.showOrderSummary }));

  /**
   * @memberof OrderReview
   * @param {object} event
   * @returns {func} this.setState
  */
  updateStateFormValues = (event) => {
    const { target } = event;
    const { name } = target;
    const { order, values } = this.state;

    return this.setState(prevState => ({
      values: {
        ...prevState.values,
        [name]: target.value
      }
    }), () => updateLocalStorageOrder(this.props.user.id, {
      ...order, ...values, [name]: target.value
    }));
  }

  /**
   * @memberof OrderReview
   * @param {string} name
   * @returns {func} this.setState
  */
  clearFormError = name => this.setState(prevState => ({
    formError: '',
    errors: {
      ...prevState.errors,
      [name]: null
    }
  }))

  /**
   * @memberof OrderReview
   * @param {string} name
   * @returns {func} this.setState
  */
  validateField = (name) => {
    const errors = syncValidate('reviewOrder')(name, this.state.values);
    const errorValue = errors || null;

    return this.setState(prevState => ({
      errors: { ...prevState.errors, [name]: errorValue }
    }));
  }

  /**
   * @memberof OrderReview
   * @returns {func} this.setState
  */
  validateForm = () => {
    const allFilled = validateRequiredFields(['deliveryPhoneNo', 'deliveryAddress'], this.state.values);
    const formErrorCount = formErrorCounter(this.state.errors);

    Object.keys(this.state.values).forEach(field => this.validateField(field));

    return !formErrorCount && allFilled;
  }

  /**
   * @memberof OrderReview
   * @param {object} event
   * @returns {func} this.toggleOrderSummary
  */
  handleOrderDetailsSubmit = () => {
    const { order, values } = this.state;

    updateLocalStorageOrder(this.props.user.id, {
      ...order, ...values
    });

    const formValid = this.validateForm();
    if (formValid) return this.toggleOrderSummary();
    return null;
  }

  /**
   * @memberof OrderReview
   * @returns {JSX} Delivery Details Component
  */
  renderInputFields = () => (
    <Fragment>
      <RenderInput
        type="tel"
        name="deliveryPhoneNo"
        id="deliveryPhoneNo"
        label="Phone Number (in the format 080xxxxxxxx)"
        required
        value={this.state.values.deliveryPhoneNo}
        placeholder=""
        handleChange={this.handleInputChange}
        handleBlur={this.handleBlur}
        handleFocus={this.handleFocus}
        meta={{ touched: true, error: this.state.errors.deliveryPhoneNo }}
      />
      <RenderInput
        type="textarea"
        name="deliveryAddress"
        id="deliveryAddress"
        label="Delivery Address"
        required
        value={this.state.values.deliveryAddress}
        placeholder=""
        handleChange={this.handleInputChange}
        handleBlur={this.handleBlur}
        handleFocus={this.handleFocus}
        meta={{ touched: true, error: this.state.errors.deliveryAddress }}
      />
    </Fragment>
  );

  /**
   * @memberof OrderReview
   * @returns {JSX} Delivery Details Component
  */
  renderDeliveryDetails = () => (
    <div>
      <h2>Delivery Details</h2>
      <hr />
      <div className="delivery-details">
        <p className="text-center" style={{ marginBottom: '30px' }}>Please fill in your details below. This is only required once.</p>
        {this.renderInputFields()}
      </div>
      <div className="control-btns">
        <Link to="/">
          <button className="btn btn-sec" type="button">Back</button>
        </Link>
        <button className="btn btn-sec" type="submit" onClick={this.handleOrderDetailsSubmit}>Next</button>
      </div>
    </div>
  );

  /**
   * @memberof OrderReview
   * @returns {JSX} OrderReview Component
  */
  render() {
    const { user, logout, isFetching } = this.props;

    if (this.state.order.meals.length === 0) return <Redirect to="/" />;
    if (this.state.showOrderSummary) return <Redirect to="/order-confirmation" />;

    return (
      <View user={user} logout={logout} type="orderReview" isFetching={isFetching}>
        <div className="order-confirmation">
          {this.renderDeliveryDetails()}
        </div>
      </View>
    );
  }
}

export default OrderReview;
