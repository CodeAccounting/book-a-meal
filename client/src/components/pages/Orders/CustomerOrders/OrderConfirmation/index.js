import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import OrderConfirmation from './OrderConfirmation';
import { logout } from '../../../../../store/operations/auth';
import { addOrder } from '../../../../../store/operations/orders';

const mapStateToProps = state => ({
  isFetching: state.isFetching
});

const mapDispatchToProps = dispatch => bindActionCreators({
  logout, addOrder,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(OrderConfirmation);
