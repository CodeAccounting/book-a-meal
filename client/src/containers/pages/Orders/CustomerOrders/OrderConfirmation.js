import { connect } from 'react-redux';
import OrderConfirmation from '../../../../components/pages/Orders/CustomerOrders/OrderConfirmation';
import { logout } from '../../../../actions/auth';
import { addOrder, editOrder } from '../../../../actions/orders';

const mapStateToProps = state => ({
  isFetching: state.isFetching
});

export default connect(mapStateToProps, { logout, addOrder, editOrder })(OrderConfirmation);
