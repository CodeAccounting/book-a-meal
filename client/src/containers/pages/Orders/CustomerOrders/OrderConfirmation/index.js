import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import OrderConfirmation from '../../../../../components/pages/Orders/CustomerOrders/OrderConfirmation';
import { logout } from '../../../../../actions/auth';
import { addOrder, editOrder } from '../../../../../actions/orders';

const mapStateToProps = state => ({
  isFetching: state.isFetching
});

const mapDispatchToProps = dispatch => bindActionCreators({
  logout, addOrder, editOrder
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(OrderConfirmation);
