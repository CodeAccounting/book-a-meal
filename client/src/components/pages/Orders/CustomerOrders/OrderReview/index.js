import { connect, } from 'react-redux';
import { bindActionCreators } from 'redux';
import OrderReview from './OrderReview';
import { logout } from '../../../../../store/operations/auth';

const mapStateToProps = state => ({
  isFetching: state.isFetching
});

const mapDispatchToProps = dispatch => bindActionCreators({
  logout
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(OrderReview);
