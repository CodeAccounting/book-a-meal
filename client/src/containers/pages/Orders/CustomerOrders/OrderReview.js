import { connect } from 'react-redux';
import OrderReview from '../../../../components/pages/Orders/CustomerOrders/OrderReview';
import { logout } from '../../../../actions/auth';

const mapStateToProps = state => ({
  isFetching: state.isFetching
});


export default connect(mapStateToProps, { logout })(OrderReview);
