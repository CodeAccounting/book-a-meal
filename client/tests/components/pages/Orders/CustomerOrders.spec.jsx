import React from 'react';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import OrderConfirmation from '../../../../src/components/pages/Orders/CustomerOrders/OrderConfirmation/OrderConfirmation';
import ConnectedOrderConfirmation from '../../../../src/components/pages/Orders/CustomerOrders/OrderConfirmation';
import OrderReview from '../../../../src/components/pages/Orders/CustomerOrders/OrderReview/OrderReview';
import ConnectedOrderReview from '../../../../src/components/pages/Orders/CustomerOrders/OrderReview';
import { customer, initialValues, localStorageOrder } from '../../../setup/data';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({ ...initialValues });

describe('Customer Orders', () => {
  beforeEach(() => {
    localStorage.setItem('bookamealorder', JSON.stringify(localStorageOrder));
  });

  describe('Order Confirmation', () => {
    it('renders correctly', () => {
      const shallowWrapper = shallow(<OrderConfirmation
        user={customer}
        logout={jest.fn()}
        addOrder={jest.fn()}
        editOrder={jest.fn()}
        isFetching={false}
      />);

      expect(shallowWrapper.find('OrderSummary').length).toBeTruthy();
    });

    it('redirects when there are no orders', () => {
      localStorage.setItem('bookamealorder', null);

      const shallowWrapper = shallow(<OrderConfirmation
        user={customer}
        logout={jest.fn()}
        addOrder={jest.fn()}
        editOrder={jest.fn()}
        isFetching
      />);

      expect(shallowWrapper.find('Redirect')).toBeTruthy();
    });

    it('renders connected component', () => {
      const comp = (
        <Provider store={store}>
          <ConnectedOrderConfirmation
            user={customer}
            isFetching={false}
          />
        </Provider>);

      const wrapper = mount(comp, rrcMock.get());

      expect(wrapper.find('OrderSummary').length).toBeTruthy();
      wrapper.unmount();
    });

    it('calls handle submit on checkout when creating a new order', () => {
      const addOrderMock = jest.fn();
      const editOrderMock = jest.fn();

      const comp = (
        <Provider store={store}>
          <OrderConfirmation
            user={customer}
            logout={jest.fn()}
            addOrder={addOrderMock}
            editOrder={editOrderMock}
            isFetching={false}
          />
        </Provider>);

      const wrapper = mount(comp, rrcMock.get()).find(OrderConfirmation);

      wrapper.find('#checkout').simulate('click');

      const storedOrder = localStorage.getItem('bookamealorder');

      expect(storedOrder).toBe(undefined);
      expect(addOrderMock).toHaveBeenCalled();
      expect(editOrderMock).not.toHaveBeenCalled();
    });

    it('calls handle submit on checkout when updating an order', () => {
      localStorage.setItem('bookamealorder', JSON.stringify({
        ...localStorageOrder,
        order: {
          ...localStorageOrder.order,
          id: 'f7247d3a-de8a-43e2-90f6-b126cd4c491c'
        }
      }));

      const addOrderMock = jest.fn();
      const editOrderMock = jest.fn();

      const comp = (
        <Provider store={store}>
          <OrderConfirmation
            user={customer}
            logout={jest.fn()}
            addOrder={addOrderMock}
            editOrder={editOrderMock}
            isFetching={false}
          />
        </Provider>);

      const wrapper = mount(comp, rrcMock.get()).find(OrderConfirmation);

      wrapper.find('#checkout').simulate('click');

      const storedOrder = localStorage.getItem('bookamealorder');

      expect(storedOrder).toBe(undefined);
      expect(editOrderMock).toHaveBeenCalled();
      expect(addOrderMock).not.toHaveBeenCalled();
    });
  });

  describe('Order Review', () => {
    it('renders correctly', () => {
      const shallowWrapper = shallow(<OrderReview
        user={customer}
        logout={jest.fn()}
        isFetching={false}
      />);

      expect(toJson(shallowWrapper)).toMatchSnapshot();
    });

    it('redirects when there are no orders', () => {
      localStorage.setItem('bookamealorder', null);

      const shallowWrapper = shallow(<OrderReview
        user={customer}
        logout={jest.fn()}
        isFetching
      />);

      expect(shallowWrapper.find('Redirect')).toBeTruthy();
    });

    it('renders connected component', () => {
      const comp = (
        <Provider store={store}>
          <ConnectedOrderReview
            user={customer}
            isFetching={false}
          />
        </Provider>);

      const wrapper = mount(comp, rrcMock.get());

      expect(toJson(wrapper)).toMatchSnapshot();
      wrapper.unmount();
    });

    it('handles input change on number and address change', () => {
      const comp = (
        <Provider store={store}>
          <OrderReview
            user={customer}
            logout={jest.fn()}
            isFetching={false}
          />
        </Provider>);

      const wrapper = mount(comp, rrcMock.get()).find(OrderReview);

      const handleInputChange = jest.spyOn(wrapper.instance(), 'handleInputChange');

      wrapper.find('input#deliveryPhoneNo').simulate('focus');
      wrapper.find('input#deliveryAddress').simulate('focus');
      wrapper.find('input#deliveryPhoneNo').simulate('change', { target: { name: 'number', value: '08122334455' } });
      wrapper.find('input#deliveryAddress').simulate('change', { target: { name: 'address', value: '2, Chruch Street, Place' } });
      wrapper.find('input#deliveryPhoneNo').simulate('blur');
      wrapper.find('input#deliveryAddress').simulate('blur');

      const storedOrder = JSON.parse(localStorage.getItem('bookamealorder'));

      expect(storedOrder.order.number).toEqual('08122334455');
      expect(storedOrder.order.address).toEqual('2, Chruch Street, Place');
      expect(handleInputChange).toHaveBeenCalled();
    });

    it('validates form and submits valid form', () => {
      const comp = (
        <Provider store={store}>
          <OrderReview
            user={customer}
            logout={jest.fn()}
            isFetching={false}
          />
        </Provider>);

      const wrapper = mount(comp, rrcMock.get()).find(OrderReview);

      const handleOrderDetailsSubmitMock = jest.spyOn(wrapper.instance(), 'handleOrderDetailsSubmit');
      const toggleOrderSummaryMock = jest.spyOn(wrapper.instance(), 'toggleOrderSummary');

      wrapper.find('input#deliveryPhoneNo').simulate('focus');
      wrapper.find('input#deliveryAddress').simulate('focus');
      wrapper.find('input#deliveryPhoneNo').simulate('change', { target: { name: 'number', value: '+2348122334455' } });
      wrapper.find('input#deliveryAddress').simulate('change', { target: { name: 'address', value: '2, Chruch Street, Place' } });
      wrapper.find('input#deliveryPhoneNo').simulate('blur');
      wrapper.find('input#deliveryAddress').simulate('blur');

      wrapper.find('button[type="submit"]').simulate('click');

      expect(handleOrderDetailsSubmitMock).toHaveBeenCalled();
      expect(toggleOrderSummaryMock).toHaveBeenCalled();
    });

    it('validates form and doesnt submits valid form', () => {
      const comp = (
        <Provider store={store}>
          <OrderReview
            user={customer}
            logout={jest.fn()}
            isFetching={false}
          />
        </Provider>);

      const wrapper = mount(comp, rrcMock.get()).find(OrderReview);

      const handleOrderDetailsSubmitMock = jest.spyOn(wrapper.instance(), 'handleOrderDetailsSubmit');
      const toggleOrderSummaryMock = jest.spyOn(wrapper.instance(), 'toggleOrderSummary');

      wrapper.find('input#deliveryPhoneNo').simulate('focus');
      wrapper.find('input#deliveryAddress').simulate('focus');
      wrapper.find('input#deliveryPhoneNo').simulate('change', { target: { name: 'deliveryPhoneNo', value: '8122334455' } });
      wrapper.find('input#deliveryAddress').simulate('change', { target: { name: 'deliveryAddress', value: '2, Chruch Street, Place' } });
      wrapper.find('input#deliveryPhoneNo').simulate('blur');
      wrapper.find('input#deliveryAddress').simulate('blur');

      wrapper.find('button[type="submit"]').simulate('click');

      expect(handleOrderDetailsSubmitMock).toHaveBeenCalled();
      expect(toggleOrderSummaryMock).not.toHaveBeenCalled();
    });
  });
});
