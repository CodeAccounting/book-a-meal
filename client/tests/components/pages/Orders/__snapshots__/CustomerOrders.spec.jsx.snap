// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Customer Orders Order Review renders connected component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(OrderReview)
    isFetching={false}
    user={
      Object {
        "email": "emiola@olasanmi.com",
        "firstname": "Emiola",
        "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
        "lastname": "Olasanmi",
        "role": "customer",
      }
    }
  >
    <OrderReview
      isFetching={false}
      logout={[Function]}
      user={
        Object {
          "email": "emiola@olasanmi.com",
          "firstname": "Emiola",
          "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
          "lastname": "Olasanmi",
          "role": "customer",
        }
      }
    >
      <View
        isFetching={false}
        logout={[Function]}
        showTime={true}
        type="orderReview"
        user={
          Object {
            "email": "emiola@olasanmi.com",
            "firstname": "Emiola",
            "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
            "lastname": "Olasanmi",
            "role": "customer",
          }
        }
      >
        <div
          className="user"
        >
          <Header
            active="orderReview"
            dateType="orderReview"
            logout={[Function]}
            showTime={true}
            type="customer"
            user={
              Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                "lastname": "Olasanmi",
                "role": "customer",
              }
            }
          >
            <div
              className="header"
            >
              <div
                className="navbar white-navbar"
              >
                <CustomerHeader
                  active="orderReview"
                  dateType="orderReview"
                  logout={[Function]}
                  showTime={true}
                  type="customer"
                  user={
                    Object {
                      "email": "emiola@olasanmi.com",
                      "firstname": "Emiola",
                      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                      "lastname": "Olasanmi",
                      "role": "customer",
                    }
                  }
                >
                  <div
                    className="page-title"
                  >
                    <h3>
                      <Link
                        href="/"
                        replace={false}
                        to="/"
                      >
                        <a
                          href="/"
                          onClick={[Function]}
                        >
                          BOOK-A-MEAL
                        </a>
                      </Link>
                    </h3>
                  </div>
                  <div
                    className="navlinks"
                  >
                    <div
                      className="user-info"
                    >
                      <Dropdown
                        content={
                          <UNDEFINED>
                            <Link
                              className="false"
                              href="/"
                              replace={false}
                              to="/"
                            >
                              Menu
                            </Link>
                            <Link
                              href="/orders"
                              replace={false}
                              to="/orders"
                            >
                              Order History
                            </Link>
                            <Link
                              href="/"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                            >
                              Logout
                            </Link>
                          </UNDEFINED>
                        }
                        toggler={
                          <div
                            className="username-circle circle-click"
                          >
                            <p
                              className="circle-click"
                            >
                              E
                              ▿
                            </p>
                          </div>
                        }
                        type="customer"
                      >
                        <div
                          className="dropdown"
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        >
                          <Link
                            className="link"
                            clickHandler={[Function]}
                            id="dropdown-toggler"
                          >
                            <button
                              className="link-btn link"
                              id="dropdown-toggler"
                              onClick={[Function]}
                            >
                              <div
                                className="username-circle circle-click"
                              >
                                <p
                                  className="circle-click"
                                >
                                  E
                                  ▿
                                </p>
                              </div>
                            </button>
                          </Link>
                          <div
                            className="dropdown-content"
                            id="dropdown-content"
                          >
                            <Link
                              className="false"
                              href="/"
                              replace={false}
                              to="/"
                            >
                              <a
                                className="false"
                                href="/"
                                onClick={[Function]}
                              >
                                Menu
                              </a>
                            </Link>
                            <Link
                              href="/orders"
                              replace={false}
                              to="/orders"
                            >
                              <a
                                href="/orders"
                                onClick={[Function]}
                              >
                                Order History
                              </a>
                            </Link>
                            <Link
                              href="/"
                              onClick={[Function]}
                              replace={false}
                              to="/"
                            >
                              <a
                                href="/"
                                onClick={[Function]}
                              >
                                Logout
                              </a>
                            </Link>
                          </div>
                        </div>
                      </Dropdown>
                      <p>
                          
                        Emiola
                      </p>
                    </div>
                  </div>
                </CustomerHeader>
              </div>
            </div>
          </Header>
          <div
            className="main-wrapper"
            id="has-modal"
          >
            <div
              className="order-confirmation"
            >
              <div>
                <h2>
                  Delivery Details
                </h2>
                <hr />
                <div
                  className="delivery-details"
                >
                  <p
                    className="text-center"
                    style={
                      Object {
                        "marginBottom": "30px",
                      }
                    }
                  >
                    Please fill in your details below. Both fields are required.
                  </p>
                  <RenderInput
                    handleBlur={[Function]}
                    handleChange={[Function]}
                    handleFocus={[Function]}
                    id="deliveryPhoneNo"
                    label="Phone Number"
                    meta={
                      Object {
                        "error": null,
                        "touched": true,
                      }
                    }
                    name="deliveryPhoneNo"
                    placeholder=""
                    required={true}
                    type="number"
                    value="+2348134567891"
                  >
                    <div
                      className="form-input"
                    >
                      <label
                        htmlFor="deliveryPhoneNo"
                      >
                        Phone Number
                        <span
                          className="danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        className="is-valid"
                        id="deliveryPhoneNo"
                        name="deliveryPhoneNo"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder=""
                        type="number"
                        value="+2348134567891"
                      />
                    </div>
                  </RenderInput>
                  <RenderInput
                    handleBlur={[Function]}
                    handleChange={[Function]}
                    handleFocus={[Function]}
                    id="deliveryAddress"
                    label="Delivery Address"
                    meta={
                      Object {
                        "error": null,
                        "touched": true,
                      }
                    }
                    name="deliveryAddress"
                    placeholder=""
                    required={true}
                    type="textarea"
                    value="4, Church Street, Yaba"
                  >
                    <div
                      className="form-input"
                    >
                      <label
                        htmlFor="deliveryAddress"
                      >
                        Delivery Address
                        <span
                          className="danger"
                        >
                          *
                        </span>
                      </label>
                      <input
                        className="is-valid"
                        id="deliveryAddress"
                        name="deliveryAddress"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder=""
                        type="textarea"
                        value="4, Church Street, Yaba"
                      />
                    </div>
                  </RenderInput>
                </div>
                <div
                  className="control-btns"
                >
                  <Link
                    href="/"
                    replace={false}
                    to="/"
                  >
                    <a
                      href="/"
                      onClick={[Function]}
                    >
                      <button
                        className="btn btn-sec"
                        type="button"
                      >
                        Back
                      </button>
                    </a>
                  </Link>
                  <button
                    className="btn btn-sec"
                    onClick={[Function]}
                    type="submit"
                  >
                    Next
                  </button>
                </div>
              </div>
            </div>
          </div>
          <Connect(Modal)
            isFetching={false}
            logout={[Function]}
            showTime={true}
            type="orderReview"
            user={
              Object {
                "email": "emiola@olasanmi.com",
                "firstname": "Emiola",
                "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                "lastname": "Olasanmi",
                "role": "customer",
              }
            }
          >
            <Modal
              isFetching={false}
              logout={[Function]}
              open={false}
              showTime={true}
              toggleModal={[Function]}
              type={null}
              user={
                Object {
                  "email": "emiola@olasanmi.com",
                  "firstname": "Emiola",
                  "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
                  "lastname": "Olasanmi",
                  "role": "customer",
                }
              }
            />
          </Connect(Modal)>
          <Footer>
            <div
              className="footer"
            >
              <p>
                © 2018. Iveren Shaguy. All images were gotten from Google and are not mine.
              </p>
            </div>
          </Footer>
        </div>
      </View>
    </OrderReview>
  </Connect(OrderReview)>
</Provider>
`;

exports[`Customer Orders Order Review renders correctly 1`] = `
<View
  isFetching={false}
  logout={[MockFunction]}
  showTime={true}
  type="orderReview"
  user={
    Object {
      "email": "emiola@olasanmi.com",
      "firstname": "Emiola",
      "id": "a09a5570-a3b2-4e21-94c3-5cf483dbd1ac",
      "lastname": "Olasanmi",
      "role": "customer",
    }
  }
>
  <div
    className="order-confirmation"
  >
    <div>
      <h2>
        Delivery Details
      </h2>
      <hr />
      <div
        className="delivery-details"
      >
        <p
          className="text-center"
          style={
            Object {
              "marginBottom": "30px",
            }
          }
        >
          Please fill in your details below. Both fields are required.
        </p>
        <React.Fragment>
          <RenderInput
            handleBlur={[Function]}
            handleChange={[Function]}
            handleFocus={[Function]}
            id="deliveryPhoneNo"
            label="Phone Number"
            meta={
              Object {
                "error": null,
                "touched": true,
              }
            }
            name="deliveryPhoneNo"
            placeholder=""
            required={true}
            type="number"
            value="+2348134567891"
          />
          <RenderInput
            handleBlur={[Function]}
            handleChange={[Function]}
            handleFocus={[Function]}
            id="deliveryAddress"
            label="Delivery Address"
            meta={
              Object {
                "error": null,
                "touched": true,
              }
            }
            name="deliveryAddress"
            placeholder=""
            required={true}
            type="textarea"
            value="4, Church Street, Yaba"
          />
        </React.Fragment>
      </div>
      <div
        className="control-btns"
      >
        <Link
          href="/"
          replace={false}
          to="/"
        >
          <button
            className="btn btn-sec"
            type="button"
          >
            Back
          </button>
        </Link>
        <button
          className="btn btn-sec"
          onClick={[Function]}
          type="submit"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</View>
`;
